docs/reference/how-to/search-speed.asciidoc

```diff
@@ -9,6 +9,27 @@ fast. In general, you should make sure that at least half the available memory
 goes to the filesystem cache so that Elasticsearch can keep hot regions of the
 index in physical memory.
 
+[discrete]
+=== Avoid page cache thrashing by using modest read ahead values on Linux
+
+Search can cause a lot of randomized read IO. When using the default
```

Comment:
Actually readahead is used regardless of the file store (even when its `niofs`) as all implementations leverage the page cache? So maybe we should simplify this paragraph and only say:

```
Search can cause a lot of randomized read I/O. When the underlying block device has a high readahead value, there may be a lot of unnecessary read I/O done.
```

Reply:
That's right! However, my understanding is (e.g. here: https://biriukov.dev/docs/page-cache/0-linux-page-cache-for-sre/) that the impact is much more pronounced when using memory mapping.

Reply:
++ The impact is pronounced due to the nature of mmap. In randomized read I/O use cases, high readahead with memory-mapped files may likely result in loading the "wrong" pages into the page cache leading to more major page faults. Do we want to touch on this?